types/config.go

```diff
@@ -13,3 +22,115 @@ func SetConfig() {
 	config.SetFullFundraiserPath(FullFundraiserPath)
 	config.Seal()
 }
+
+const (
+	// base chain ids for xpla network
+	XplaBaseDenom    = "axpla"
+	XplaDisplayDenom = "xpla"
+
+	// base chain ids for xpla network
+	XplaMainnetChainID   = "dimension_37"
+	XplaTestnetChainID   = "cube_47"
+	XplaHypercubeChainID = "hypercube_270"
+
+	// default value
+	DefaultBaseDenom    = "axpla"
+	DefaultDisplayDenom = "xpla"
+	DefaultChainID      = "test_9999"
+)
+
+// cosmosEVMActivators defines a map of opcode modifiers associated
+// with a key defining the corresponding EIP.
+var cosmosEVMActivators = map[int]func(*vm.JumpTable){
+	0o000: eips.Enable0000,
+	0o001: eips.Enable0001,
+	0o002: eips.Enable0002,
+}
+
+// EVMOptionsFn defines a function type for setting app options specifically for
+// the Cosmos EVM app. The function should receive the chainID and return an error if
+// any.
+type EVMOptionsFn func(string) error
+
+// NoOpEVMOptions is a no-op function that can be used when the app does not
+// need any specific configuration.
+func NoOpEVMOptions(_ string) error {
+	return nil
+}
+
+var sealed = false
+
+// ChainsCoinInfo is a map of the chain id and its corresponding EvmCoinInfo
+// that allows initializing the app with different coin info based on the
+// chain id
+var ChainsCoinInfo = map[string]evmtypes.EvmCoinInfo{
+	XplaMainnetChainID: {
+		Denom:         XplaBaseDenom,
+		ExtendedDenom: XplaBaseDenom,
+		DisplayDenom:  XplaDisplayDenom,
+		Decimals:      evmtypes.EighteenDecimals,
+	},
+	XplaTestnetChainID: {
+		Denom:         XplaBaseDenom,
+		ExtendedDenom: XplaBaseDenom,
+		DisplayDenom:  XplaDisplayDenom,
+		Decimals:      evmtypes.EighteenDecimals,
+	},
+	XplaHypercubeChainID: {
+		Denom:         XplaBaseDenom,
+		ExtendedDenom: XplaBaseDenom,
+		DisplayDenom:  XplaDisplayDenom,
+		Decimals:      evmtypes.EighteenDecimals,
+	},
+	DefaultChainID: {
+		Denom:         DefaultBaseDenom,
+		ExtendedDenom: DefaultBaseDenom,
+		DisplayDenom:  DefaultDisplayDenom,
+		Decimals:      evmtypes.EighteenDecimals,
+	},
+}
+
+// EvmAppOptions allows to setup the global configuration
+// for the Cosmos EVM chain.
+func EvmAppOptions(chainID string) error {
+	if sealed {
+		return nil
+	}
+
+	id := strings.Split(chainID, "-")[0]
+	coinInfo, found := ChainsCoinInfo[id]
+	if !found {
+		coinInfo, _ = ChainsCoinInfo[DefaultChainID]
```

Comment:
Is there a reason to set it to DefaultChainID? It seems confusing. How about error handling?

Reply:
There's an issue where the application doesn't start when an arbitrary chain ID is used during local testing. This also happens `go test` and e2e tests. So, I've added this.